### ISSUE TITLE
data-api documentation lacking

### ISSUE THREAD

**Issue by pashute** [rstaib/jquery-bootgrid#239]
There is a problem in the documentation. It is not clear which attributes to use for the data-api when under the column section. Which are data-column-nnn  and which are data-nnn? 

For example the visible attribute is data-visible and NOT data-column-visible, although it is documented under the columns section. 

So some way to give the data-api naming is needed. 

At least tell us where to look in the code to find this?  I can read javascript fairly well but could not find the definitions. Are they in the css?

**Comment by FiroStefani** [comment]
I like this plugin and am using it but I also found the data-api documentation frustrating at times. It is unfortunate because the design is a good and the code is working well for me. There is a JQuery trap to be aware of also. See http://stackoverflow.com/questions/22753629/jquery-get-html-5-data-attributes-with-hyphens-and-case-sensitivity 

TL;DR: 
- Do **not** write: `data-camelCase=`
- Instead write: `data-camel-case=`

These are the attributes for columns I've gleaned from the Javascript source. Look for calls to `$this.data()` if you want to explore further:
- `data-column-id` // not `data-id` as the documentation suggests
- `data-identifier`
- `data-converter`
- `data-header-align` // [sic], `data-headerAlign` _will fail_ in JQuery 1.5+
- `data-css-class` // as above
- `data-header-css-class` // as above
- `data-formatter`
- `data-order`
- `data-searchable`
- `data-sortable`
- `data-visible`
- `data-visible-in-selection` // as above
- `data-width`

**Comment by WhiteKarma** [comment]
What about setting case insensitive searches with the data-api? Could anyone solve this? i tried a _lot_ of combinations with no success. I think the plugin is beautiful and it works well but you have to be some kind of wizard to guess how it works and not to get totally frustrated loosing tons of hours with the lack of examples in the documentation. It is its only problem in my view. It is a problem easy to solve and it would save a lot of people a lot of headaches. It is really a pity that so much good work is so hard to figure out and possibly leads lots of people to other plugins because of loosing so much time even with misleading information.

Thanks in advance

**Comment by philwig** [comment]
I hit the data-headerAlign issue and could not fix it by the obvious data-header-align fix, which for whatever jQuery 2.\* reason doesn't work the way you'd expect. Simply changing the Bootgrid code to make this all lower case one-word does the job: data-headerAlign in the markup and data-headeralign in the JS.

**Comment by guffzilla** [comment]
I just spent 45 minutes trying to center-align my column headers because the documentation is incorrect.
headerAlign does not work and it wasn't very obvious to me that this should be written as data-header-align 

Luckily I stumbed across this post - so thank you for the clarification
